src/components/Context/CardListProvider.tsx

```diff
@@ -0,0 +1,56 @@
+import { createContext, useState } from 'react'
+import React from 'react'
+import { Card } from '../types'
+
+type CardContextType = {
+	cardList: Card[]
+	setCardList: React.Dispatch<React.SetStateAction<Card[]>>
+	isAdding: boolean
+	setIsAdding: React.Dispatch<React.SetStateAction<boolean>>
+	editingCard: Card
+	setEditingCard: React.Dispatch<React.SetStateAction<Card>>
+	isSecondSide: boolean
+	setIsSecondSide: React.Dispatch<React.SetStateAction<boolean>>
+}
+const defaultState = [{ title: '', value: '' }]
+const defaultContext: CardContextType = {
+	cardList: defaultState,
+	setCardList: () => {},
+	isAdding: false,
+	setIsAdding: () => {},
+	editingCard: {
+		title: '',
+		value: '',
+	},
+	setEditingCard: () => {},
+	isSecondSide: false,
+	setIsSecondSide: () => {},
+}
+
+const CardContext = createContext<CardContextType>(defaultContext)
+
+const CardListProvider = ({ children }) => {
+	const [editingCard, setEditingCard] = useState<Card>({
```

Comment:
**nit**: this should not be a part of the provider state, as it's related to the single card